instrumentation/rmi/javaagent/rmi-javaagent.gradle

```diff
@@ -8,7 +8,14 @@ muzzle {
 
 task "rmic", dependsOn: testClasses {
   def clazz = 'rmi.app.ServerLegacy'
-  String command = """rmic -g -keep -classpath ${sourceSets.test.output.classesDirs.asPath} -d ${buildDir}/classes/java/test ${clazz}"""
+
+  // Try one level up too in case java.home refers to jre directory inside jdk directory
+  def rmicBinaryPath = ['/bin/rmic', '/../bin/rmic'].findResult {
+    def path = new File(System.getProperty("java.home"), it).getAbsoluteFile()
+    path.isFile() ? path.toString() : null
+  } ?: "rmic"
+
+  String command = """$rmicBinaryPath -g -keep -classpath ${sourceSets.test.output.classesDirs.asPath} -d ${buildDir}/classes/java/test ${clazz}"""
```

Comment:
Why do we do this during task creation? Shouldn't this be in `doLast`? How does it work at all??

Reply:
Not actually sure why it is done this way, just changed it because I do not keep a JDK in `PATH`, and because this may cause tooling from different JDKs to be mixed.

But it does indeed look like `doLast` was intended here, because if you run the tests from a clean state, the `rmic` runs before everything else, and thus sees no input (which would be produced by test compile task, which runs later). However, the test actually passes without the files generated by `rmic` being present at all, which raises the question of why even have them.

So it might actually make sense to throw this task away, since all clean builds up to now have always passed without the `rmic` outputs anyway, so functionally nothing would change. Alternatively, would have to change the test so it actually makes use of the stubs (and thus fails without their presence) if the intent was to make sure instrumentation doesn't break anything with stubs, but that seems like a full task and not a simple fix as was intended with this PR.